csharp/src/IceLocatorDiscovery/PluginI.cs

```diff
@@ -202,15 +201,14 @@ public void
             }
         }
 
-        public override Task<Ice.Object_Ice_invokeResult>
-        IceInvokeAsync(byte[] inParams, Current current)
+        public async ValueTask<OutputStream> DispatchAsync(InputStream inputStream, Current current)
         {
-            lock (this)
-            {
-                var request = new Request(this, current.Operation, current.IsIdempotent, inParams, current.Context);
-                Invoke(null, request);
-                return request.Task;
-            }
+            var requestFrame = new IncomingRequestFrame(inputStream, current);
+            var request = new Request(this, current.Operation, current.IsIdempotent, requestFrame.TakePayload(),
+                current.Context);
+            Invoke(null, request);
+            IncomingResponseFrame incoming = await request.Task.ConfigureAwait(false);
```

Comment:
maybe rename `incoming` to `responseFrame`

Reply:
Renamed to incomingResponseFrame.